catalog/dags/common/utils.py

```diff
@@ -0,0 +1,69 @@
+import functools
+from typing import Any
+
+from common.constants import SQL_INFO_BY_MEDIA_TYPE
+
+
+def setup_kwargs_for_media_type(
+    values_by_media_type: dict[str, Any], kwarg_name: str
+) -> callable:
```

Comment:
Question regarding the implementation: is the customisable `kwarg_name` a byproduct of existing implementations? I'm wondering specifically why we don't always use the `media_info` kwarg name. Is it because existing implementations use a different name and you wanted to avoid making changes to those names?

Edit: Never mind. I misunderstood how this function was used, but after reading `setup_sql_info_for_media_type` at the end, I get now why `kwarg_name` is configurable. It's dependent on what the "values" are in `values_by_media_type` :+1: Disregard this comment. Leaving it intact just in case someone else runs into the same misconception about how this works.